### PR TITLE
PRMP-568 - Modify the Front End to handle new poll based download logic

### PR DESCRIPTION
**Pull request by joefong-nhs**

### CODE REVIEW

File: app/cypress/e2e/0-ndr-core-tests/gp_user_workflows/download_lloyd_george_workflow.cy.js [hunk 1]
```diff
@@ -59,10 +59,17 @@ describe('GP Workflow: View Lloyd George record', () => {
...
+                statusCode: 200,
+                body: { jobId: 'test-jobId' },
+            });
+        }).as('documentManifestPost');
+
         cy.intercept('GET', '/DocumentManifest*', (req) => {
             req.reply({
                 statusCode: 200,
-                body: baseUrl + '/browserconfig.xml', // uses public served file in place of a ZIP file
+                body: { jobStatus: 'Completed', url: baseUrl + '/browserconfig.xml' }, // uses public served file in place of a ZIP file
```
Comment (NogaNHS) [inline app/cypress/e2e/0-ndr-core-tests/gp_user_workflows/download_lloyd_george_workflow.cy.js:72]: Could you add a test case for if the jobStatus is still pending and the url is empty?

Comment (joefong-nhs) [reply app/cypress/e2e/0-ndr-core-tests/gp_user_workflows/download_lloyd_george_workflow.cy.js]: Thanks, I have added the pending to the base test case. The intercept will now respond with "Processing" and an empty url for a few times before it responds with "Complete"

File: app/src/helpers/requests/getPresignedUrlForZip.ts [hunk 2]
```diff
@@ -11,30 +17,85 @@ type Args = {
...
-type GetPresignedUrl = {
-    data: string;
+type GetRequestArgs = {
+    jobId: string;
+    baseUrl: string;
+    baseHeaders: AuthHeaders;
+};
+
+const ThreePendingErrorMessage =
+    'Document Manifest api call responded with "Pending" for 3 attempts.';
```
Comment (steph-torres-nhs) [inline app/src/helpers/requests/getPresignedUrlForZip.ts:27]: Maybe a different message, "Failed to initiate download"?

File: app/cypress/e2e/0-ndr-core-tests/gp_user_workflows/download_lloyd_george_workflow.cy.js (hunk 1 above)
Comment (joefong-nhs) [reply app/cypress/e2e/0-ndr-core-tests/gp_user_workflows/download_lloyd_george_workflow.cy.js]: Also add a test case for getting "Pending" 3 times and display an error page

File: app/src/helpers/requests/getPresignedUrlForZip.ts (hunk 2 above)
Comment (joefong-nhs) [reply app/src/helpers/requests/getPresignedUrlForZip.ts]: thanks! fixed this just now

**Review by NogaNHS — APPROVED** [review]
